pulsar-metadata/src/test/java/org/apache/pulsar/metadata/TestZKServer.java

```diff
@@ -25,33 +25,31 @@
 import java.io.IOException;
 import java.io.InputStreamReader;
 import java.io.OutputStream;
-import java.net.InetSocketAddress;
+import java.lang.reflect.Field;
 import java.net.Socket;
+import java.util.Properties;
 
-import java.nio.charset.StandardCharsets;
-
+import lombok.SneakyThrows;
 import lombok.extern.slf4j.Slf4j;
 
 import org.apache.commons.io.FileUtils;
-import org.apache.zookeeper.KeeperException;
 import org.apache.zookeeper.server.ContainerManager;
-import org.apache.zookeeper.server.NIOServerCnxnFactory;
-import org.apache.zookeeper.server.Request;
-import org.apache.zookeeper.server.RequestProcessor;
 import org.apache.zookeeper.server.ServerCnxnFactory;
-import org.apache.zookeeper.server.SessionTracker;
 import org.apache.zookeeper.server.ZooKeeperServer;
+import org.apache.zookeeper.server.ZooKeeperServerMain;
+import org.apache.zookeeper.server.embedded.ExitHandler;
+import org.apache.zookeeper.server.embedded.ZooKeeperServerEmbedded;
 import org.assertj.core.util.Files;
+import org.junit.platform.commons.util.ReflectionUtils;
```

Comment:
There is also an issue with this line. It is failing checkstyle because of this recent PR https://github.com/apache/pulsar/pull/20611

Reply:
this is TestZKServer, it is testing code, so it is fine to use JUnit utilities IMHO

Reply:
btw, I have removed it